### DISCUSSION TITLE
Display Current Server Utilization for Deployed Containers

### DISCUSSION THREAD

**Discussion by Spark61** [coollabsio/coolify#2039] [Feature Requests]
I propose adding a new feature to Coolify that allows users to view the current server utilization for all deployed containers. This feature would include a graph for each container, showing its real-time resource usage (CPU, memory, etc.). Additionally, it would be beneficial to display the overall server utilization on the dashboard page.

This enhancement will provide users with better insights into their container performance and server load, facilitating more efficient resource management.

**Benefits:**
- Enables proactive resource management.
- Enhances visibility into application performance and health.
- Improves user experience by providing essential information directly on the dashboard.

Looking forward to feedback and suggestions on this proposal.

**Comment by nilsonmolina** [comment]
This is my most requested feature for Coolify. Without this, I'm considering using alternatives that do support this, something like dokploy.

How are others currently handling this in Coolify?

**Comment by andrasbacsai** [comment]
I would like to make this available soon. 

I have several solutions in my mind, but unsure which one should be the best.

Real-time monitoring is not that hard. To have stored timeseries data of the server and each container is a bit harder.

**Reply by marko-hologram** [reply]
Can cAdvisor be used for this? https://github.com/google/cadvisor

It seems like a good fit and it actually offers an API that can be used to get data that it collects about running containers.

**Reply by vniehues** [reply]
I think it would be a great start to show the current usage to give some insight into the server resouces
Adding history could be a great addition but might as well be another feature coming later down the line

**Reply by ladderschool** [reply]
> Can cAdvisor be used for this? https://github.com/google/cadvisor
> 
> It seems like a good fit and it actually offers an API that can be used to get data that it collects about running containers.

cAdvisor seems to be a perfect API that could be integrated in a tab for each container/service. What's more, perhaps you could do monitoring and alerts where you get an e-mail if a container hits a certain memory or bandwidth limit. Very useful in production.

**Reply by marko-hologram** [reply]
@ladderschool this server metrics feature seems to be released already in beta 298 https://github.com/coollabsio/coolify/releases/tag/v4.0.0-beta.298

Haven't tested the feature and not sure how it's implemented, but it seems to exist now.

**Reply by OSZII** [reply]
@marko-hologram do you see anything I'm currently on beta 323 and can't find a metrics tab

**Reply by marko-hologram** [reply]
@OSZII https://github.com/coollabsio/coolify/releases/tag/v4.0.0-beta.312 seems metrics were disabled in beta 312 because of some issues.

**Comment by nhCoder** [comment]
Any Update on this this feature request?

**Comment by kitety** [comment]
I also want this function, can it be like dokploy?

**Comment by vaibhavc2** [comment]
This is a really important thing. There should be an overall Monitoring system as well as a Monitoring dashboard for each of the deployed container service in each project. This is the most IMPORTANT feature needed ASAP.

**Comment by theDanielJLewis** [comment]
I would also love to see this because `docker stats` and especially `htop` don't help us know _which_ container is using the resources.

**Reply by LukeTarr** [reply]
This is not a perfect work around because it will get rid of cached containers, but if you set the container name for each of your apps to the application name under Advanced -> Custom Container Name and then restart everything you’ll be able to see which app is under what load with docker stats or htop.

I believe this forces Coolify to re-build every single time you want to restart but it’s worth it to me to be able to monitor my server resource utilization.

I’m also looking into using Dozzle which gives logs and resource monitor for Docker and has a script on Coolify to install it! I’ll update this when I do!

**Reply by theDanielJLewis** [reply]
Unfortunately, that's not possible with WordPress stacks.

**Reply by LukeTarr** [reply]
Is there like a prebuilt WordPress click to deploy you’re using? 

I know nothing about Wordpress but if you can get it running in a docker container then you can just use coolify to deploy from repo (public or private) and then the setting in advanced will be there to choose a custom container name.

I also set up a “Dozzle with Auth” instance from service tab and it basically is a web app version of docker stats with a little more info like logs from the containers and total system resources and total current usage so it’s decent for this but you still need to container name to tell which app is which.

**Reply by theDanielJLewis** [reply]
Yes, I'm using the Coolify built-in stack for WordPress. It creates two containers: WordPress (Apache) and either MariaDB or MySQL. There is no option to change the container names.

**Reply by LukeTarr** [reply]
Gotcha yeah the build in stacks don’t offer the same level of customization.

it’s kinda a pain but you can host your own projects from a repo and then you’ll be able to rename them. Otherwise I’m not sure how to do it in the built-in ones.

**Comment by nexus9111** [comment]
Hello. Any update ?

**Comment by 0xtlt** [comment]
![CleanShot 2024-11-27 at 10 19 42@2x](https://github.com/user-attachments/assets/4c7f8e0e-ac98-4003-8aa7-4f5f698b5bec)

Seems well deployed :)

@nexus9111, what do you need exactly?

**Reply by hlassiege** [reply]
Where does this screenshot comes from ?
I don't have this screen on my instance

Edit : never mind, it's in servers => configuration

**Reply by towfiqi** [reply]
@0xtlt I don't see this in my instance under Servers > Configuration. I am using [v4.0.0-beta.373](https://github.com/coollabsio/coolify/releases/tag/v4.0.0-beta.373) Do I need to enable something somewhere to get the metrics?

**Reply by 0xtlt** [reply]
![CleanShot 2024-11-29 at 08 36 12@2x](https://github.com/user-attachments/assets/3721646c-8102-4479-a71f-f280486fee14)

Yes you have to enable Sentinel + Metrics @towfiqi

**Reply by towfiqi** [reply]
Thank You! :)

**Reply by Abel1011** [reply]
It would be good to include information on the use of the disk.

**Comment by aminhir** [comment]
You are great guys. Please add a page as Monitoring that shows all containers, like various types of charts show how much each container is using.

**Reply by gevera** [reply]
This is something I am looking forward as well

**Comment by fishnux** [comment]
I'm looking for disk usage metrics

**Comment by Tam2** [comment]
Can we add a mini version of these metrics to the server select page when adding a new application or viewing which server an application is on 

When managing a large number of servers and distributing apps across them it's useful to know at a glance the metrics (cpu, ram, storage e & number or apps deployed)

Helps to more evenly distribute apps across them when adding new apps or moving apps between servers

**Comment by ikwerre-dev** [comment]
Is there an API to access these details?

**Comment by SultanDino** [comment]
Is the disk usage displayed somewhere in the dashboard? I can't find it anywhere.
